### DISCUSSION TITLE
How to include my own tcp to be visualised in rviz and gazebo?

### DISCUSSION THREAD

**Discussion by T-BonesLek** [UniversalRobots/Universal_Robots_ROS2_Driver#946] [Q&A]
To be able to update my new urdf.xacro for ur5 cb3 with my custom gripper included I must perform "colcon build" on the source directory such that  ur_robot_driver/urdf/ur.urdf.xacro is updated and be executable. I get lots of errors when trying to build the repo out of the box. Is there another way to include my gripper than trying to change UR's working Driver package.

**Comment by RobertWilbrandt** [comment]
The normal workflow for this is to create a custom package with your own URDF that includes and instantiates [ur_macro](https://github.com/UniversalRobots/Universal_Robots_ROS2_Description/blob/rolling/urdf/ur_macro.xacro). The driver supports this by letting you specify the [`description_package`](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_robot_driver/launch/ur_control.launch.py#L453) and [`description_file`](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/blob/main/ur_robot_driver/launch/ur_control.launch.py#L461) in the launch file. We are currently creating a tutorial for this [here](https://github.com/UniversalRobots/Universal_Robots_ROS_Tutorials/pull/12), even if it is not finished you can take a look if that is helpful for you.

If you have further compilation problems with the robot please file an issue containing your workflow and the relevant build logs.
